attestation-service/verifier/src/snp/mod.rs

```diff
@@ -253,16 +276,26 @@ pub(crate) fn parse_tee_evidence(report: &AttestationReport) -> TeeEvidenceParse
     claims_map as TeeEvidenceParsedClaim
 }
 
+fn get_common_name(cert: &x509::X509) -> Result<String> {
+    let mut entries = cert.subject_name().entries_by_nid(Nid::COMMONNAME);
+
+    if let Some(e) = entries.next() {
+        assert_eq!(entries.count(), 0);
+        return Ok(e.data().as_utf8()?.to_string());
+    }
+    Err(anyhow!("No CN found"))
+}
```

Comment:
Try to use `let Some() ... else` for better reading.

```suggestion
fn get_common_name(cert: &x509::X509) -> Result<String> {
    let mut entries = cert.subject_name().entries_by_nid(Nid::COMMONNAME);

    let Some(e) = entries.next() else {
        bail!("No CN found");
    }
    
    if entries.count() != 0 {
        ...
    }
    
    Ok(...)
}
```